File: Arrays/867. Transpose Matrix.py
# Url - https://leetcode.com/problems/transpose-matrix/

class Solution:
    def transpose(self, matrix: List[List[int]]) -> List[List[int]]:
        nrows = len(matrix)
        ncols = len(matrix[0])
        
        ansMatrix = [[None] * nrows for _ in range(ncols)]
        
        for i in range (0, nrows):
            for j in range(0, ncols):
                ansMatrix[j][i] = matrix[i][j]
                
        return ansMatrix
        

File: Arrays/867. Transpose Matrix.cpp
// Url - https://leetcode.com/problems/transpose-matrix/

class Solution {
public:
    vector<vector<int>> transpose(vector<vector<int>>& matrix) {
        int nrows = matrix.size();
        int ncols = matrix[0].size();
        
        vector<vector<int>> ansMatrix(ncols, vector(nrows, INT_MIN));
        
        for(int i = 0; i < nrows; i++)
        {
            for(int j = 0; j < ncols; j++)
            {
                ansMatrix[j][i] = matrix[i][j];
            }
        }
        
        return ansMatrix;
    }
};